### PR TITLE
Move ubo WebGL Extension to engine

### PR DESCRIPTION
**Pull request by RaananW**
@sebavan - Does this influence ThinEngine in any way? should it move there?

### CODE REVIEW

**Review by deltakosh — APPROVED** [review]

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

**Comment by bjsplat** [comment]
Snapshot stored with reference name:
refs/pull/15598/merge

Test environment: 
https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/15598/merge/index.html

To test a playground add it to the URL, for example:

https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/15598/merge/index.html#WGZLGJ#4600

Links to test babylon tools with this snapshot:

https://playground.babylonjs.com/?snapshot=refs/pull/15598/merge
https://sandbox.babylonjs.com/?snapshot=refs/pull/15598/merge
https://gui.babylonjs.com/?snapshot=refs/pull/15598/merge
https://nme.babylonjs.com/?snapshot=refs/pull/15598/merge

To test the snapshot in the playground with a playground ID add it after the snapshot query string:

https://playground.babylonjs.com/?snapshot=refs/pull/15598/merge#BCU1XR#0

File: packages/dev/core/src/Engines/engine.ts [hunk 1]
```diff
@@ -26,6 +26,7 @@ import "./Extensions/engine.cubeTexture";
 import "./Extensions/engine.renderTarget";
 import "./Extensions/engine.renderTargetTexture";
 import "./Extensions/engine.renderTargetCube";
+import "../Engines/Extensions/engine.uniformBuffer";
```
Comment (Popov72) [inline packages/dev/core/src/Engines/engine.ts:29]:
```suggestion
import "./Extensions/engine.uniformBuffer";
```

**Review by Popov72 — APPROVED** [review]

**Comment by bjsplat** [comment]
WebGL2 visualization test reporter:

https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/15598/merge/testResults/webgl2playwright/index.html

**Comment by bjsplat** [comment]
Visualization tests for WebGPU (Experimental)

https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/15598/merge/testResults/webgpuplaywright/index.html

**Comment by sebavan** [comment]
It does impact, but not all projects are using UBO so it should not be moved in but wondering if there is a way maybe in the UBO ctor to detect the lack of support and log a warning import ?

**Pull request merged**